### DISCUSSION TITLE
Let player select color of (parts of) vehicles

### DISCUSSION THREAD

**Discussion by hurzelchen** [voxeltycoon/issues#494] [General]
It would be a nice addition to give the player the possibility to select the color of parts of the vehicles. This would allow to distinguish between vehicles on different routes more easily.

Maybe limit the selection to the same colors as the company colors so that the exact shades could be defined in the models.

**Comment by victorleblais** [comment]
I'm 100% on board with this.

I know that many people prefer to keep the "realistic" aspect of the liveries, but having coloring as an option would be great.

Company coloring is apparently planned according to Trello: https://trello.com/c/ZRcAuztt/57-company-colours

But your suggestion to take it one step further is cool. I always loved seeing my fleet of color-coordinated vehicles in OpenTTD.

That would allow several options:
- Don't paint vehicles
- Paint vehicles using the company color
- Paint vehicles using the route color (which could be selected by the player).

And I guess similar to how you select railcars in the train building windows, you could select which units you want to paint and which color. I'll try to whip up a mockup UI soon.

**Comment by victorleblais** [comment]
This feature is partially implemented in 0.85.

Using the asset editor, you can create liveries: basically another color scheme for that asset.
There's even a new slider for "Company Color" which allows to tint the company color with various level of saturation and brightness, it's pretty neat.

However, I'm not closing this issue since there is currently no way in-game to choose between different liveries. You can only use the default/first one.

Let's hope that'll come soon, since most of the work has been done.
